core/src/engine/EngineMultiRobot.cc

```diff
@@ -1115,9 +1115,8 @@ namespace jiminy
                     while (tNext - t > EPS)
                     {
                         /* Adjust stepsize to end up exactly at the next breakpoint,
-                           prevent steps larger than dtMax, and make sure that dt is
-                           multiple of TELEMETRY_TIME_DISCRETIZATION_FACTOR whenever
-                           it is possible, to reduce rounding errors of logged data. */
+                           prevent steps larger than dtMax, trying to reach multiples of
+                           STEPPER_MIN_TIMESTEP whenever possible. */
```

Comment:
@duburcqa this deserves some thoughts.
You implemented a mechanism to synchronized simulation timestep with the log unit. Now the log unit can change. I can keep the mechanism as is, but this means changing the precision of the logging mechanism changes the result of the simulation: arguably this will be a very minor change, but the idea is still weird. I would rather remove this mechanism: either always synchronized at 1us as an arbitrary precision threshold (given that having timestamps that vary at a 1e-9 level is not really relevant for robotics), or remove this altogether. I think you did this for unit tests, in which case we can have unit test run with a 1e-8 precision for time without this mechanism and see if that works. 

What do you think?

Reply:
Good point. I think it is still interesting to sync the timestep with 1us as an arbitrary precision. By doing this, the simulation is not affected by the logging time precision,  time logging at the default precision is more accurate, people using higher accuracy should not really see any difference, and it feels good to know there is a bit of order in the big mess of the timesteps x)

Reply:
Totally agree. Even though 1us is arbitrary, from a robotics stand point it really makes sense.